integration-tests/tests/common/mod.rs

```diff
@@ -76,17 +84,29 @@ impl IntegrationTest {
             .arg(self.code)
             .arg("--shim")
             .arg(self.shim)
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
             .spawn()
             .expect("failed to run the test");
 
+        let input = cmd.stdin.as_mut().unwrap();
```

Comment:
```suggestion
        let input = cmd.stdin.unwrap();
```

I don't understand the purpose of the `.as_mut()` here...

Reply:
Also, I think we can get rid of the input temporary variable.

Reply:
You are right, we don't need the temporary variable, I got rid of it now.
The value stored in cmd is getting moved and is resulting in an error when I try getting the ecode without as_mut.
I am not entirely sure how creating a mutable reference with as_mut() is solving the problem